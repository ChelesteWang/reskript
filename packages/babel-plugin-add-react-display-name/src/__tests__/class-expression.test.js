const runSuite = require('./suite');

runSuite('class-expression');
