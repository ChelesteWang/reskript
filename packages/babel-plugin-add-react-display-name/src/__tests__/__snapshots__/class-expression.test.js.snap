// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class expression anonymous class 1`] = `
"// anonymous class

/* eslint-disable */
import { Component } from 'react';
const Foo = class extends Component {
  render() {
    return /*#__PURE__*/React.createElement(\\"div\\", null);
  }

};
Foo.displayName = \\"Foo\\";"
`;

exports[`class expression anonymous class assigned to object property 1`] = `
"// anonymous class assigned to object property

/* eslint-disable */
import { Component } from 'react';
const global = {
  namespace: {}
};
global.namespace.Foo = class extends Component {
  render() {
    return /*#__PURE__*/React.createElement(\\"div\\", null);
  }

};
global.namespace.Foo.displayName = \\"Foo\\";"
`;

exports[`class expression anonymous class assigned to variable 1`] = `
"// anonymous class assigned to variable

/* eslint-disable */
import { Component } from 'react';
let Foo;
Foo = class extends Component {
  render() {
    return /*#__PURE__*/React.createElement(\\"div\\", null);
  }

};
Foo.displayName = \\"Foo\\";"
`;

exports[`class expression multiple react imports 1`] = `
"// multiple react imports

/* eslint-disable */
import React from 'react';
import { Component } from 'react';
const Foo = class Bar extends Component {
  render() {
    return /*#__PURE__*/React.createElement(\\"div\\", null);
  }

};
Foo.displayName = \\"Bar\\";"
`;

exports[`class expression named class 1`] = `
"// named class

/* eslint-disable */
import { Component } from 'react';
const Foo = class Bar extends Component {
  render() {
    return /*#__PURE__*/React.createElement(\\"div\\", null);
  }

};
Foo.displayName = \\"Bar\\";"
`;

exports[`class expression named class assigned to object property 1`] = `
"// named class assigned to object property

/* eslint-disable */
import { Component } from 'react';
const global = {
  namespace: {}
};
global.namespace.Foo = class Bar extends Component {
  render() {
    return /*#__PURE__*/React.createElement(\\"div\\", null);
  }

};
global.namespace.Foo.displayName = \\"Bar\\";"
`;

exports[`class expression nested in child scope as a variable 1`] = `
"// nested in child scope as a variable

/* eslint-disable */
import { Component } from 'react';

const factory = () => {
  const Foo = class extends Component {
    render() {
      return /*#__PURE__*/React.createElement(\\"div\\", null);
    }

  };
  Foo.displayName = \\"Foo\\";
  return Foo;
};"
`;

exports[`class expression nested in child scope as an assignment 1`] = `
"// nested in child scope as an assignment

/* eslint-disable */
import { Component } from 'react';

const initialize = () => {
  window.Foo = class extends Component {
    render() {
      return /*#__PURE__*/React.createElement(\\"div\\", null);
    }

  };
  window.Foo.displayName = \\"Foo\\";
};"
`;

exports[`class expression returned from function 1`] = `
"// returned from function

/* eslint-disable */
import { Component } from 'react';

const factory = () => class extends Component {
  render() {
    return /*#__PURE__*/React.createElement(\\"div\\", null);
  }

};"
`;

exports[`class expression without super class 1`] = `
"// without super class

/* eslint-disable */
const Foo = class {
  render() {
    return /*#__PURE__*/React.createElement(\\"div\\", null);
  }

};"
`;
